.github/workflows/unittest.yml

```diff
@@ -0,0 +1,34 @@
+name: Unittest
```

Comment:
Is there any advantage to combining them? On the contrary, the separation is very useful when the actions are actually failing, and the log needs to be troubleshooted. Also, a failure blocks subsequent actions so troubleshooting multiple failing actions won't follow agile model.
I'll investigate further on combining/ separation and best practices. Thanks for the insights Alberto.

Reply:
That's a fair point, I agree is useful to have them run in parallel.

My concerns are 

1. Locking to many resources
2. Eventually having a crazy number of actions

Not sure what are the best practices, I would love to learn more if you find something in your investigation

Reply:
I've concluded my research. It seems like there isn't clear consensus (maybe because the concept is new? **maybe I didn't search efficiently**? could be both).
I've also looked at workflows of most starred python github repos. They seem to agree with the "minimal" action philosophy which I'll be sticking to.

About your points:
1. The resources issue; let's talk hardware vs software:
  - Hardware:
    Seems to not be an issue, the virtual machines spin up so fast, it's enough to optimize the code running.

  - Software (code of the action itself):
     It doesn't really make any difference running two jobs in the same workflow or in two separate ones.
 
2. The "too much actions" issue:
    First we need to define how much is too much, I've seen repos with +10 actions.
    It's very easy to merge two flows later on. And if it turns out merging flows is hard, it is probably because they require different setups and thus different "flows".

Let me know what you think. @albert17

Reply:
I'd personally would have 2 actions, one for unittest and another for lint (pytest+black). 

But your approach of keeping it as it is now works for me, as you said we can always merge if required. We can see how it evolves and we can discuss in the future if required.

Thanks for looking into this! :)